### ISSUE TITLE
Away status no longer shown

### ISSUE THREAD

**Issue by grawity** [weechat/weechat#430]
```--> :kornbluth.freenode.net 311 grawity evilgrawity grawity wolke.nullroute.eu.org * :Mantas Mikulėnas <grawity@gmail.com>
--> :kornbluth.freenode.net 319 grawity evilgrawity :#archers @#nullroute #systemd @#archlinux   
--> :kornbluth.freenode.net 312 grawity evilgrawity kornbluth.freenode.net :Frankfurt, Germany
--> :kornbluth.freenode.net 301 grawity evilgrawity :I'm a bot.
--> :kornbluth.freenode.net 671 grawity evilgrawity :is using a secure connection
--> :kornbluth.freenode.net 317 grawity evilgrawity 791 1431939044 :secondsidle, signon time
--> :kornbluth.freenode.net 330 grawity evilgrawity grawity :is logged in as
--> :kornbluth.freenode.net 318 grawity evilgrawity :End of /WHOIS list.

**Comment by flashcode** [comment]
Hi,
This is shown for me in server buffer when I `/whois` someone who is away.
Do you have some scripts loaded, that could disturb?
Or did you add some nicks with `/notify` ? (check `/notify`)

**Comment by grawity** [comment]
Yeah, I have one nick in the /notify list for freenode.

But actually, I just realized what triggers this. I had a /query open for the user. So when I ran /whois, the "is away" message went to the query buffer, but the rest of whois was shown in the server buffer as usual.

**Comment by flashcode** [comment]
Yes, and this is the expected behavior: the IRC server sends the same message inside the whois and when you talk to someone who is away. So in both cases, WeeChat will display that in private buffer (if opened).
The only way to improve things here would be to redirect the whole output of whois (to use redirects), and display it with "keeping in mind" we are in a /whois output, ie everything must go to the server buffer and not the private buffer.
Anyway this becomes a feature request, so I change the type of issue.

**Comment by mikko-s-hirvonen** [comment]
A workaround for this problem is to have the whole output of /whois sent to the query buffer:
/set irc.msgbuffer.whois private
